pkg/olm/clusterserviceversion.go

```diff
@@ -123,6 +123,49 @@ func (builder *ClusterServiceVersionBuilder) GetAlmExamples() (string, error) {
 	return "", fmt.Errorf("%s not found in given csv named %v", almExamples, builder.Definition.Name)
 }
 
+// IsSucceeded check if the csv is Succeeded.
+func (builder *ClusterServiceVersionBuilder) IsSucceeded() (bool, error) {
+	if valid, err := builder.validate(); !valid {
+		return false, err
+	}
+
+	glog.V(100).Infof("Verify %s csv in %s namespace is Succeeded",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	phase, err := builder.GetPhase()
+
+	if err != nil {
+		return false, fmt.Errorf("failed to get phase value for %s clusterserviceversion in %s namespace due to %w",
+			builder.Definition.Name, builder.Definition.Namespace, err)
+	}
+
+	if phase == "Succeeded" {
```

Comment:
I think you can just return here. IMO error should be reserved for things that go wrong while performing an operation. 
```suggestion
        return phase == "Succeeded", nil
```